web/components/logentry/weight.jsx

```diff
@@ -0,0 +1,140 @@
+import CurrentEntryActions from '../../actions/current-entry-actions';
+import formUtils from '../../utils/form-utils';
+import PropTypes from 'prop-types';
+import React from 'react';
+import TextBox from '../controls/text-box.jsx';
+
+import {
+	Col,
+	ControlLabel,
+	FormGroup,
+	Radio,
+	Row
+} from 'react-bootstrap';
+
+class Weight extends React.Component {
+	constructor() {
+		super();
+		this.onValueChanged = this.onValueChanged.bind(this);
+	}
+
+	onValueChanged(e) {
+		const weight = this.props.entry.weight
+			? Object.assign({}, this.props.entry.weight)
+			: {};
+
+		if (e.target.id) {
+			switch (e.target.id) {
+				case 'weight':
+					weight.amount = formUtils.tryReturnAsNumber(e.target.value);
+					break;
+			}
+		} else {
+			switch (e.target.name) {
+				case 'weight':
+					weight.correctness = e.target.value;
+					break;
+
+				case 'trim':
+					weight.trim = e.target.value;
+					break;
+			}
+		}
+
+		CurrentEntryActions.doPartialUpdate({ weight: weight });
+	}
+
+	render() {
+		const weight = this.props.entry.weight || {};
+
+		return (
+			<div>
+				<Row>
+					<Col xs={12}>
+						<h4>Weight</h4>
+					</Col>
+				</Row>
+				<TextBox
+					controlId="weight"
+					name="weight"
+					label="Weight"
+					unit="lbs"
+					value={ weight.amount }
+					onChange={ this.onValueChanged }
+					validations={{
+						isInteger: true,
```

Comment:
This does not need to be an integer! There's no reason someone couldn't wear 8.5lbs or some other fractional value.